ansible/templates/usb/scripts/prompt-config.sh.j2

```diff
@@ -1586,6 +1633,26 @@ or you rely only on the dhcp.\n\n"
 
 	promptnet "Enter the default gateway IP" "$headnode_default_gateway" "headnode_default_gateway"
 	headnode_default_gateway="$val"
+	# if headnode_default_gateway is outside the admin subnet,
+	# we need to ask for the admin gw
+	if ! ip_from_subnet "$headnode_default_gateway" "$admin_ip" "$admin_netmask"; then
+		if [ -z "$admin_gateway" ]; then
+			# set default value
+			admin_gateway=$(num_to_ip $(($NET_NUM + 1)) && echo $ip_addr)
+		fi
+		while true; do
```

Comment:
Impossible. Function `ip_from_subnet` uses other functions that use global variables. And therefore it cannot be called recursively.